core/impl/src/main/java/com/blazebit/persistence/impl/function/datetime/epoch/OracleEpochFunction.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2014 - 2017 Blazebit.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.blazebit.persistence.impl.function.datetime.epoch;
+
+import com.blazebit.persistence.spi.FunctionRenderContext;
+import com.blazebit.persistence.spi.TemplateRenderer;
+
+public class OracleEpochFunction extends EpochFunction {
+
+    private final TemplateRenderer paramRenderer;
+
+    public OracleEpochFunction() {
+        super("extract(second from (cast(?1 as timestamp) - to_date('1970-01-01', 'yyyy-mm-dd'))) " +
```

Comment:
This implementation seems wrong. I read that `cast(?1 as timestamp) - to_date('01-Jan-1970', 'dd-Mon-yyyy')) *24*60*60*1000` gives the millis so you could simplify this a lot. Here the site: http://stackoverflow.com/questions/30211623/how-to-get-the-epoch-in-pl-sql